### PR TITLE
Docker support

### PR DESCRIPTION
**Pull request by gentlecat**
*Documentation still needs to be written.*

### CODE REVIEW

**Comment by gentlecat** [comment]
Also need to figure out how to do configuration properly.

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+# Node
+node_modules
+
+# Logs
+*.log
+pip-log.txt
+pip-delete-this-directory.txt
+
+# Application data
+/data
```
Comment (mayhem) [inline .dockerignore]: is this supposed to be absolute?

File: docker/nginx_custom/nginx.conf [hunk 2]
```diff
@@ -0,0 +1,24 @@
+worker_processes 4;
```
Comment (mayhem) [inline docker/nginx_custom/nginx.conf]: Any thoughts on how to do docker configs for dev vs deployment?

**Comment by mayhem** [comment]
Overall this looks pretty good to me. Clearly we still need to learn more about different configurations (dev vs production), but this certainly looks good enough to try and do a test deploy.

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      - .:/code
+      - ./data/app:/data
+    ports:
+      - "3031:3031"
+    depends_on:
+      - db
+      - memcached
+
+  nginx:
+    restart: always
```
Comment (mayhem) [inline docker-compose.yml]: Why are the other modules not marked at restart?

**Comment by mayhem** [comment]
Aside from the comments already posted and the gulp/static stuff, this is looking VERY GOOD.

File: .dockerignore (hunk 1 above)
Comment (gentlecat) [inline .dockerignore]: Yes. It's in the root of the context (git repo). https://docs.docker.com/engine/reference/builder/#dockerignore-file

**Comment by gentlecat** [comment]
This branch has been merged into `beta`.

**Pull request closed without merging**